Find a functionally equivalent code:android.appwidget.AppWidgetHost.allocateAppWidgetId:COMMENT
Method Modifier: public      
Comment:/**
 * Get a appWidgetId for a host in the calling process.
 *
 * @return a appWidgetId
 */

@@ -1,9 +1,6 @@
 {
     try {
-        if (mPackageName == null) {
-            mPackageName = mContext.getPackageName();
-        }
-        return sService.allocateAppWidgetId(mPackageName, mHostId, mContext.getUserId());
+        return sService.allocateAppWidgetId(mContext.getOpPackageName(), mHostId);
     } catch (RemoteException e) {
         throw new RuntimeException("system server dead?", e);
     }

